### PR TITLE
Ye Old Ugly Boolean, Yanked

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin/interface.hpp [hunk 1]
```diff
@@ -179,14 +179,14 @@ QP CDECL principia__CelestialFromParent(Plugin const* const plugin,
 // Calls |plugin->NewBodyCentredNonRotatingFrame| with the arguments given.
 // |plugin| must not be null.  The caller gets ownership of the returned object.
 extern "C" DLLEXPORT
-Transforms<Barycentric, Rendering, Barycentric>* CDECL
+RenderingTransforms* CDECL
 principia__NewBodyCentredNonRotatingTransforms(Plugin const* const plugin,
```
Comment (eggrobin) [inline ksp_plugin/interface.hpp]: Fits on previous line.

File: ksp_plugin/interface.hpp [hunk 2]
```diff
@@ -179,14 +179,14 @@ QP CDECL principia__CelestialFromParent(Plugin const* const plugin,
...
                                                int const reference_body_index);
 
 // Calls |plugin->NewBarycentricRotatingFrame| with the arguments given.
 // |plugin| must not be null.  The caller gets ownership of the returned object.
 extern "C" DLLEXPORT
-Transforms<Barycentric, Rendering, Barycentric>* CDECL
+RenderingTransforms* CDECL
 principia__NewBarycentricRotatingTransforms(Plugin const* const plugin,
```
Comment (eggrobin) [inline ksp_plugin/interface.hpp]: same.

File: ksp_plugin/interface.cpp [hunk 3]
```diff
@@ -254,14 +254,14 @@ QP principia__CelestialFromParent(Plugin const* const plugin,
           ToXYZ(result.velocity().coordinates() / (Metre / Second))};
 }
 
-Transforms<Barycentric, Rendering, Barycentric>*
+RenderingTransforms*
 principia__NewBodyCentredNonRotatingTransforms(Plugin const* const plugin,
```
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: fits on previous line.

File: ksp_plugin/interface.cpp [hunk 4]
```diff
@@ -254,14 +254,14 @@ QP principia__CelestialFromParent(Plugin const* const plugin,
...
                                                int const reference_body_index) {
   return CHECK_NOTNULL(plugin)->
       NewBodyCentredNonRotatingTransforms(reference_body_index).release();
 }
 
-Transforms<Barycentric, Rendering, Barycentric>*
+RenderingTransforms*
 principia__NewBarycentricRotatingTransforms(Plugin const* const plugin,
```
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: same.

File: ksp_plugin/celestial.hpp [hunk 5]
```diff
@@ -39,18 +39,18 @@ class Celestial {
...
+  not_null<Trajectory<Barycentric>*> mutable_history() override;
 
   // Both accessors require |is_initialized()|.
-  Trajectory<Barycentric> const& prolongation() const;
-  not_null<Trajectory<Barycentric>*> mutable_prolongation();
+  Trajectory<Barycentric> const& prolongation() const override;
+  not_null<Trajectory<Barycentric>*> mutable_prolongation() override;
 
   // Both accessors require |is_initialized()|.  In addition the first one
   // requires |has_prediction()|.
```
Comment (eggrobin) [inline ksp_plugin/celestial.hpp]: Do we want to repeat the contract documentation three times?

File: physics/transforms_body.hpp [hunk 6]
```diff
@@ -217,22 +223,22 @@ Transforms<FromFrame, ThroughFrame, ToFrame>::BarycentricRotating(
...
   transforms->second_ =
-      [to_primary_trajectory, to_secondary_trajectory](
+      [&primary, &secondary, to_trajectory](
           Instant const& t,
           DegreesOfFreedom<ThroughFrame> const& through_degrees_of_freedom,
           Trajectory<ThroughFrame> const* trajectory) ->
       DegreesOfFreedom<ToFrame> {
     DegreesOfFreedom<ToFrame> const& last_primary_degrees_of_freedom =
-        to_primary_trajectory().last().degrees_of_freedom();
+        (primary.*to_trajectory)().last().degrees_of_freedom();
```
Comment (eggrobin) [inline physics/transforms_body.hpp]: put the result of `LazyTrajectory` in a local variable, if it changes between calls here some threading magic is happening that violates the contract anyway.

File: ksp_plugin/interface.hpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/interface.hpp]: Done.

File: ksp_plugin/interface.hpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/interface.hpp]: Done.

File: ksp_plugin/interface.cpp (hunk 4 above)
Comment (pleroy) [inline ksp_plugin/interface.cpp]: Done.

File: ksp_plugin/celestial.hpp (hunk 5 above)
Comment (pleroy) [inline ksp_plugin/celestial.hpp]: Removing the contract from the interface, it doesn't make much sense there since there is no notion of `is_initialized`.

**Pull request merged**